controllers/machinehealthcheck_controller.go

```diff
@@ -254,10 +255,50 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 
 		if annotations.IsPaused(cluster, t.Machine) {
 			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
-		} else {
+		} else if m.Spec.RemediationTemplate != nil {
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Remediation template not found %q", m.Spec.RemediationTemplate)
+			}
+
+			generateTemplateInput := &external.GenerateTemplateInput{
+				Template:    from,
+				TemplateRef: m.Spec.RemediationTemplate,
+				Namespace:   t.Machine.Namespace,
+				ClusterName: t.Machine.ClusterName,
+				OwnerRef:    cloneOwnerRef,
+				Labels:      nil,
+			}
+			to, err := external.GenerateTemplate(generateTemplateInput)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create template for remediation request %q", m.Spec.RemediationTemplate)
+			}
+
+			to.SetName(t.Machine.Name)
+
+			// Create the external clone.
+			if err := r.Client.Create(context.Background(), to); err != nil {
```

Comment:
I think we need to check if one already exists before creating. Otherwise we will fail on `Create` repeatedly

Reply:
How about checking in line 258 like this:
`else if m.Spec.RemediationTemplate != nil && t.Machine.Status.RemadiationRef == nil` 

This prevent `Create` to fail, but we need to make sure that adding the `RemediationRef` not failing. Like you refer comment below. If we fail to add reference the `Remediation Request` needs to be deleted as well.